packages/dashboard-frontend/src/containers/IdeLoader.tsx

```diff
@@ -422,6 +423,13 @@ class IdeLoaderContainer extends React.PureComponent<Props, State> {
     }
     const workspace = this.props.allWorkspaces.find(workspace => workspace.id === cheWorkspace.id);
     if (workspace && workspace.ideUrl) {
+      try {
+        if (workspace.isDevWorkspace) {
+          await injectKubeConfig(workspace.namespace, workspace.id);
```

Comment:
@JPinkney I have several questions:
1. Why should we do it each time when we open IDE?
This solution is going to do it each time when open running IDE from the Dashboard 

2. Could we move this functionality on the server side? I am not sure that it is the right place to do it (client-side of UD).

@akurinnoy WDYT?

Reply:
@olexii4 I have the same thoughts about it.

Reply:
> 1. Why should we do it each time when we open IDE?
> This solution is going to do it each time when open running IDE from the Dashboard

If the home directory is not persisted then the kubeconfig would only be injected the first time and then never injected ever again. It calls injectKubeConfig each time but if a kubeconfig has already been injected its essentially a no-op.

> 2. Is it possible to move this part to the devworkspace-operator side to inject kubeconfig there?

The devworkspace operator has no way of figuring out who the user is or what their token is that's why it was decided to put this mechanism on the dashboard. Similiar-ish to how kubeconfig injection works with Web Terminal Operator except this solution is more generic, works will all editors and doesn't require che-machine-exec. Other then that the mechanism of action is the exact same as how the OpenShift Console does it.

> 2a. Could we move this functionality to the server-side at least?

We _might_ be able to technically move it to the server side but what's the gain? We still need something to initiate the call to the backend when the pod is fully started or just constantly wait until the pod status is started.

Reply:
If the home directory is not persistent, we need to inject the kube-config on the workplace start each time. 

This solution does not work for such actions as Run Workspace in background and Restart Workspace. This can lead to errors in the case of single-host.

I would suggest transferring this logic to the startWorkspace action (devworkspaces redux store) to have this implementation in a single place.

Reply:
> If the home directory is not persistent, we need to inject the kube-config on the workplace start each time.

That's how its currently implemented

> This solution does not work for such actions as Run Workspace in background and Restart Workspace. This can lead to errors in the case of single-host.

Can you elaborate? Why wouldn't it work?

> I would suggest transferring this logic to the startWorkspace action (devworkspaces redux store) to have this implementation in a single place.

You can't inject the kubeconfig until all the pods are ready. At best I might be able to watch the pod from the startWorkspace action and inject after its in a running state

Reply:
> You can't inject the kubeconfig until all the pods are ready. At best I might be able to watch the pod from the startWorkspace action and inject after its in a running state

It is the best solution(watch the pod from the startWorkspace action and inject after it's in a running state).

We have an example here:
https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts#L315-L336

Reply:
> Can you elaborate? Why wouldn't it work?

Use single-host mode. Imagine two tabs that will be open in the browser(workspaces list and workspace editor). Click on Restart action.
![2022-01-04_18_44_09](https://user-images.githubusercontent.com/6310786/148098563-a565fd2c-1dfc-48b2-8831-3398f3c31dc1.png)

Reply:
"Watch the pod from the startWorkspace action" should fix it.